### DISCUSSION TITLE
Detailed explains about tmTheme?

### DISCUSSION THREAD

**Discussion by linrongbin16** [microsoft/vscode-discussions#1029] [Extension Development QnA] [answered]
The core theme system of vscode is from the TextMate, e.g. the tm theme, see: https://macromates.com/manual/en/themes.

This system is also been used in sublime text, see: https://www.sublimetext.com/docs/color_schemes_tmtheme.html.

But I searched a lot about the tmTheme rules, still have a lot confuse about those details, for example I have below python code:

```python

import typing

class MyClass:
  def __init__(self, age: typing.Optional[int] = None) -> None:
    self.age = age


mine = MyClass(10)
```

In such an easy code, I am confused what is `entity.name.class`, what is `variable.other.constant`, etc.

Thus I am not sure what color should I use when I try to create/customize my theme.

Is there any detailed docs about tmTheme?

**Comment by tamuratak** [comment, marked as answer]
You can use VS Code's built-in scope inspector tool:

- https://code.visualstudio.com/api/language-extensions/syntax-highlight-guide#scope-inspector

For naming conventions, you can refer to:

- https://macromates.com/manual/en/language_grammars#naming_conventions

VS Code's python syntax comes from:

- https://github.com/MagicStack/MagicPython/tree/master
- https://github.com/MagicStack/MagicPython/blob/master/misc/scopes would help.

**Reply by linrongbin16** [reply]
thank you for sharing!
